src/pages/search/advanced_search/FundingSourceFilter.tsx

```diff
@@ -0,0 +1,69 @@
+import { Autocomplete } from '@mui/material';
+import { useQuery } from '@tanstack/react-query';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { fetchFundingSources } from '../../../api/cristinApi';
+import { ResultParam } from '../../../api/searchApi';
+import { AutocompleteTextField } from '../../../components/AutocompleteTextField';
+import { setNotification } from '../../../redux/notificationSlice';
+import { FundingSource } from '../../../types/project.types';
+import { getLanguageString } from '../../../utils/translation-helpers';
+
+export const FundingSourceFilter = () => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const paramName = ResultParam.FundingSource;
+  const history = useHistory();
+  const searchParams = new URLSearchParams(history.location.search);
+  const currentSearchTerm = searchParams.get(paramName) || '';
+  const currentFundingSourceIdentifier = `https://api.dev.nva.aws.unit.no/cristin/funding-sources/${currentSearchTerm}`;
+
+  const fundingSourcesQuery = useQuery({
+    queryKey: ['fundingSources'],
+    queryFn: fetchFundingSources,
+    onError: () => dispatch(setNotification({ message: t('feedback.error.get_funding_sources'), variant: 'error' })),
+    staleTime: Infinity,
+    cacheTime: 1_800_000, // 30 minutes
+  });
+  const fundingSourcesList = fundingSourcesQuery.data?.sources ?? [];
+
+  const handleChange = (selectedValue: FundingSource | null) => {
+    if (selectedValue) {
+      const value = selectedValue.id.split('/').pop() ?? '';
+      searchParams.set(paramName, value);
+    } else {
+      searchParams.delete(paramName);
+    }
+
+    history.push({ search: searchParams.toString() });
+  };
+
+  return (
+    <Autocomplete
+      sx={{ minWidth: '15rem' }}
+      disabled={!fundingSourcesQuery.data}
+      value={fundingSourcesList.find((source) => source.id === currentFundingSourceIdentifier) || null}
+      onChange={(_, newValue) => {
+        handleChange(newValue);
+      }}
+      options={fundingSourcesList}
+      getOptionLabel={(option) => getLanguageString(option.name)}
+      renderOption={(props, option) => (
+        <li {...props} key={option.identifier}>
+          {getLanguageString(option.name)}
+        </li>
+      )}
+      renderInput={(params) => (
+        <AutocompleteTextField
+          isLoading={fundingSourcesQuery.isLoading}
+          {...params}
+          variant="outlined"
+          placeholder={t('search.search_for_funder')}
+          showSearchIcon={!currentSearchTerm}
+          multiline
```

Comment:
Usikker på om denne burde være `multiline`, da det kan bli litt funky med lange verdier.
<img width="713" alt="Screenshot 2024-02-14 at 13 27 40" src="https://github.com/BIBSYSDEV/NVA-Frontend/assets/53341294/3cb5fcdb-3f3a-4517-b42a-a9fe7a36421a">

Reply:
Enig, tror multiline kan bli rotete. Iallefall når det kommer flere filter under, som da begynner å hoppe rundt? Hva med å gjøre den bredere, og/eller gjemme overflow bak "..."?

Reply:
Enig i at det er litt funky, men ser ikke noen veldig god løsning egentlig 🤔 Vi har samme (multiline) for søkene på institusjon og underenhet også, og syns det er ok for nå, da jeg liker at hele navnet vises også... Men vi må nok finne en bedre løsning for dette en gang